presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -2228,6 +2238,65 @@ public void dropView(ConnectorSession session, SchemaTableName viewName)
         return views.build();
     }
 
+    @Override
+    public Optional<ConnectorMaterializedViewDefinition> getMaterializedView(ConnectorSession session, SchemaTableName viewName)
+    {
+        requireNonNull(viewName, "viewName is null");
+        Optional<Table> table = metastore.getTable(viewName.getSchemaName(), viewName.getTableName());
+        if (table.isPresent() && MetastoreUtil.isPrestoMaterializedView(table.get())) {
+            try {
+                JsonCodec<ConnectorMaterializedViewDefinition> codec = JsonCodec.jsonCodec(ConnectorMaterializedViewDefinition.class);
+                return Optional.of(codec.fromJson(decodeViewData(table.get().getViewOriginalText().get())));
+            }
+            catch (IllegalArgumentException e) {
+                throw new PrestoException(INVALID_VIEW, "Invalid materialized view JSON", e);
+            }
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public void createMaterializedView(ConnectorSession session, ConnectorTableMetadata viewMetadata, ConnectorMaterializedViewDefinition viewDefinition, boolean ignoreExisting)
+    {
+        if (isExternalTable(viewMetadata.getProperties())) {
+            throw new PrestoException(INVALID_TABLE_PROPERTY, "Specifying external location for materialized view is not supported.");
+        }
+        Table basicTable = prepareTable(session, viewMetadata, MATERIALIZED_VIEW);
+        JsonCodec<ConnectorMaterializedViewDefinition> codec = JsonCodec.jsonCodec(ConnectorMaterializedViewDefinition.class);
+        Map<String, String> parameters = ImmutableMap.<String, String>builder()
+                .putAll(basicTable.getParameters())
+                .put(PRESTO_MATERIALIZED_VIEW_FLAG, "true")
+                .build();
+        Table viewTable = Table.builder(basicTable)
+                .setParameters(parameters)
+                .setViewOriginalText(Optional.of(encodeViewData(codec.toJson(viewDefinition))))
+                .setViewExpandedText(Optional.of("/* Presto Materialized View */"))
+                // Use PrestoTableType.MANAGED_TABLE temporarily until table type MATERIALIZED_VIEW is supported by Metastore
+                 .setTableType(MANAGED_TABLE)
```

Comment:
The TableType field set here will pass all the way down to Metastore. So I swap MATERIALIZED_VIEW out here.

Feeling it's more safe to reset it here, given I found a fair number of callsites of Table.getTableType(), ex. FileHiveMetastore.  Might need to check thoroughly all of the places and reset otherwise.

Reply:
James, can you help double check I don't need to change ThriftHiveMetastore (or InMemoryHiveMetastore) in presto repo in addition to PrismHiveMetastore? I think the former will set and pass TableType.MATERIALIZED_VIEW to Metastore, outside FB.

Reply:
That is the correct understanding